Wastewater_Seq_analysis/WWP_seq_initialize_analysis.sh

```diff
@@ -0,0 +1,111 @@
+2) Setup directory structure for initiating Wastewater sequencing run analysis and any downstream analysis.
+3) Generate ncbi submission folder that can be directly used for uploading files to NCBI.
+
+Usage:
+sh WWP_seq_initialize_analysis.sh <wastewater sequencing run_name> | tee -a WWP_seq_initialize_analysis.log
+Last updated on June 16,2022
+"
+###########################
+
+echo $USAGE
+
+run_name=$1
+log_files=/Volumes/NGS/Analysis/covidseq/$run_name/covidseq_output/Logs_Intermediates
+
+if [ ! -d "$log_files/VirusDetection" ]
+then
+    echo "$(date) : Fastq generation step is not yet completed for run $run_name" 
+    exit 1
+fi
+
+echo "$(date) : Fastq generation step is completed for run $run_name."
+
+sample_sheet=$(ls $log_files/SampleSheetValidation/SampleSheet_Intermediate.csv | head -n 1)
+if [ ! -n "$sample_sheet" ]
+then
+    echo "$(date) : The sample sheet for run $run_name does not exist. Expected $sample_sheet"
+    exit 1
+fi
+echo "$(date) : The sample sheet for run $run_name is $sample_sheet"
+
+#Check whether sample sheet contains any wastewater samples. This will mostly be true.            
+if grep -nqi 'Wastewater' $sample_sheet
+then
+    echo "$(date) : Some wastewater samples found in $run_name"
+    else 
+        echo "$(date) : No wastewater samples found in $run_name .. exiting"
+        exit 1
+fi
+
+#Setting directory structure for the new WW run
+analysis_dir='/Volumes/IDGenomics_NAS/wastewater_sequencing'
+echo "$(date) : Analysis directory is $analysis_dir."
+
+if [ -d "$analysis_dir/$run_name" ]
+then
+    echo "$(date) : $run_name run folder already exists in the $analysis_dir .. exiting"
+    exit 1
+fi
+
+echo "$(date) : $run_name run folder was not found in the $analysis_dir"
+echo "$(date) : Creating $run_name run folder and sub directories for storing raw fastq files and other downstream analysis"
+mkdir -p $analysis_dir/$run_name/{raw_data,ncbi_submission,fasta,analysis,failed_samples,results}
+
+#Output directory paths
+#echo "$(date) : Raw fastq files will be stored in $analysis_dir/$run_name/raw_data"
+#echo "$(date) : Fasta files will be stored in $analysis_dir/$run_name/fasta"
+#echo "$(date) : Bioinformatics analysis will be stored in $analysis_dir/$run_name/analysis"
+
+#Get fastq files for each wastewater sample in the new run
+echo "$(date) : Generating list of wastewater samples from the run sample sheet. Used to fetch matching fastq files in the next step"
+grep -ni 'Wastewater' $sample_sheet | cut -f 2 -d ',' > $analysis_dir/$run_name/${run_name}_wastewater_sample_list.csv
+                    
+fastq_dir=$log_files/FastqGeneration
+echo "$(date) : Copying fastq files from $fastq_dir to wastewater sequencing analysis directory to run with viralrecon or Cecret"
+find $fastq_dir -type f -name '*.fastq.gz' -print0 | grep -zf $analysis_dir/$run_name/${run_name}_wastewater_sample_list.csv | xargs -r0 cp -t $analysis_dir/$run_name/raw_data
```

Comment:
This will go faster by using parallel instead of xargs (because the files will copied in parallel instead of sequentially)

```
find $fastq_dir -type f -name '*.fastq.gz' -print0 | grep -zf $analysis_dir/$run_name/${run_name}_wastewater_sample_list.csv | parallel cp {} $analysis_dir/$run_name/raw_data
```